notebook/static/notebook/js/menubar.js

```diff
@@ -190,99 +187,59 @@ define([
-        });
-        this.element.find('#run_all_cells_below').click(function () {
-            that.notebook.execute_cells_below();
-        });
-        this.element.find('#to_code').click(function () {
-            that.notebook.to_code();
-        });
-        this.element.find('#to_markdown').click(function () {
-            that.notebook.to_markdown();
-        });
-        this.element.find('#to_raw').click(function () {
-            that.notebook.to_raw();
-        });
-        
-        this.element.find('#toggle_current_output').click(function () {
-            that.notebook.toggle_output();
-        });
-        this.element.find('#toggle_current_output_scroll').click(function () {
-            that.notebook.toggle_output_scroll();
-        });
-        this.element.find('#clear_current_output').click(function () {
-            that.notebook.clear_output();
-        });
+             
+        var id_actions_dict = {
+            '#search_and_replace' : 'ipython.search-and-replace-dialog',
+            '#save_checkpoint': 'ipython.save-notebook',
+            '#restart_kernel': 'ipython.restart-kernel',
+            '#int_kernel': 'ipython.interrupt-kernel',
+            '#cut_cell': 'ipython.cut-selected-cell',
+            '#copy_cell': 'ipython.copy-selected-cell',
+            '#delete_cell': 'ipython.delete-cell',
+            '#undelete_cell': 'ipython.undo-last-cell-deletion',
+            '#split_cell': 'ipython.split-cell-at-cursor',
+            '#merge_cell_above': 'ipython.merge-selected-cell-with-cell-before',
+            '#merge_cell_below': 'ipython.merge-selected-cell-with-cell-after',
+            '#move_cell_up': 'ipython.move-selected-cell-up',
+            '#move_cell_down': 'ipython.move-selected-cell-down',
+            '#toggle_header': 'ipython.toggle-header',
+            '#toggle_toolbar': 'ipython.toggle-toolbar',
+            '#insert_cell_above': 'ipython.insert-cell-before',
+            '#insert_cell_below': 'ipython.insert-cell-after',
+            '#run_cell': 'ipython.execute-in-place',
+            '#run_cell_select_below': 'ipython.run-select-next',
+            '#run_cell_insert_below': 'ipython.execute-and-insert-after',
+            '#run_all_cells': 'ipython.run-all-cells',
+            '#run_all_cells_above': 'ipython.run-all-cells-above',
+            '#run_all_cells_below': 'ipython.run-all-cells-below',
+            '#to_code': 'ipython.change-selected-cell-to-code-cell',
+            '#to_markdown': 'ipython.change-selected-cell-to-markdown-cell',
+            '#to_raw': 'ipython.change-selected-cell-to-raw-cell',
+            '#toggle_current_output': 'ipython.toggle-output-visibility-selected-cell',
+            '#toggle_current_output_scroll': 'ipython.toggle-output-scrolling-selected-cell',
+            '#clear_current_output': 'ipython.clear-output-selected-cell',
+        };
+
+        var keys = Object.keys(id_actions_dict);
+        for(var i in keys){
```

Comment:
Sorry. `for(...in)` is just never safe. It might as well not exist. You have to use

```javascript
Object.keys(obj).map(function (key) {...});
```

or filter on `obj.hasOwnProperty(key)`.

[some illustration](http://nbviewer.ipython.org/gist/minrk/e00c51d06b6479e366c0)